ADR/ADR034-api-v2-for-forms.md

```diff
@@ -0,0 +1,408 @@
+# ADR034: API v2 for Forms
+
+Date: 2024-08-19
+
+## Status
+
+Accepted
+
+## Context
+
+Currently forms are modelled in the forms-api app, not the forms-admin app, however the forms-admin does most of the work manipulating forms. This slows down development when we need to make changes to how forms are represented, such as when we want to add new features.
+
+## Decision
+
+We've decided to create a new set of API endpoints in the forms-api app with a simpler interface, and move the form model.
+
+## Consequences
+
+- developers will be able to make more changes to forms-admin without having to also change forms-api
+- forms-api will be simpler
+- form objects will not be validated by forms-api
+
+## Appendix: Original request for comment
+
+Below is reproduced the RFC that was shared with the team to help make this decision:
+
+### Introduction
+
+Design goals:
+
+- Allow moving model logic from forms-api to forms-admin
+- Change building block of forms from pages to steps
+- Improve developer experience when making changes to form schema
+
+Non-goals:
+
+- Changing API endpoints unrelated to forms/pages
+- Supporting external consumers
+
+Design considerations:
+
+- Treat forms as opaque JSON documents
+- No endpoints for pages/conditions etc, wire format is basically just form snapshot
+- Form is made up of steps as per https://github.com/alphagov/forms/discussions/174
+- No validation of form steps (?)
+- Better handling of live/draft/archived (?)
+- Don’t use ActiveResource (?)
+
+What does that unlock?
+
+Can move functionality from api to admin incrementally (?)
+Steps abstraction makes JSON document structure less complex (?)
+Eventually less duplication of code between API and admin, more responsive admin app
+
+### Future directions
+
+The aim of these changes is to enable future changes, rather than being an ideal endpoint.
+
+This RFC tries to support any of the following future changes to the forms-api app:
```

Comment:
Is this an RFC? Or an ADR?

Reply:
It's an ADR, that includes the text of an RFC...

Reply:
I thought about changing the text of the RFC originally so it said ADR instead, but when I tried it I found it made things weird because an ADR is a different thing to an RFC. So I left it as is.

Reply:
I'd just mention that the ADR is based on an RFC...

Reply:
@chao-xian so you'd prefer it to be rewritten?

Reply:
not a rewrite just a mention at the top or on the PR